posts/development-notes-notification.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Development Notes: Notification"
+description: "Usage instructions for @tournant/notification."
+date: 2020-01-30
+tags: dev-notes
+layout: layouts/post.njk
+---
+
+Our latest component gives you the ability to communicate timely information with your users. You can notify them about stuff and other stuff. It’s [@tournant/notification](https://www.npmjs.com/package/@tournant/notification).
+
+Let’s write some words about the process of building it and where it is or isn’t useful.
+
+## A notification is a notification is a notification?
+
+I am glad you asked, headline. The answer is: Nah.
+
+Let’s imagine the real world for a second. You are sitting on the edge of silver future, wondering how it might be. Suddenly you washing machine yells: «I’m done!», interrupting your thoughts completely. A bit overzealous.
+
+Now, on the other hand a self-driving car that does not warn you, loud and clear, that it is about to make a break, breaks and slams your face into your breakfast is tiny bit too shy.
+
+I guess the point I’m trying to make is: It depends. Some notifications need to communicated immediately and with urgency. While some can wait.
+
+Our notification component knows this and can fit into any situation. In technical terms, the component is a _live region_ that can have a role of either `status` or `alert`.
+
```

Comment:
+ "Actually, [there are more](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#Preferring_specialized_live_region_roles). But in order to not make things more complicated, and to take account of the fact that in reality in vast majorities status and alert are in use\*, tournant notification is limited to these two."

\* Undocumented claim! But I think so

Reply:
Actually, it might make sense if at least `log` would be a an option. 

To which sentence does «undocumented claim» relate to?

Reply:
"in reality in vast majorities status and alert are in use". I'm meta-commenting my own comment, none of the original text.

Yeah, maybe log being the third. In an ideal world we would have some AT support data for it here: https://a11ysupport.io/tech/aria/log_role :/

Reply:
I looked into some log implementations and they are incompatible. Would need to be another component.

Reply:
Ok, my bad. Then let's leave it at status/alert

Reply:
I also didn’t know this. But a log is a container in which messages are displayed (via a list or something) whereas status/alert are single messages.

Reply:
Updated the article with the link to MDN :)